### PR TITLE
chore(pymongo): fix async client RuntimeErrors

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

**Note: This PR fixes an unreleased bug a releasenote is not required.**

Resolves: https://github.com/DataDog/dd-trace-py/issues/16449

Fixes `RuntimeError: coroutine ignored GeneratorExit` in pymongo async instrumentation when `AsyncMongoClient.close()` is called while operations are in flight.

**Root Cause:**
Async wrappers use a synchronous context manager (`with trace_cmd(...)`) inside async functions. When a coroutine suspends inside the `with` block and is then closed, `Span.__exit__` doesn't re-raise `GeneratorExit`, causing the RuntimeError.

**Solution:**
Created `async_trace_cmd`, an `@asynccontextmanager` wrapper around `trace_cmd` that properly handles `GeneratorExit`. Updated `trace_async_socket_command` and `trace_async_socket_write_command` to use `async with async_trace_cmd(...)`.

## Testing

Manually verified. This edge case is difficult to test reliably due to timing requirements and Python version-specific behavior (reported in Python 3.14).

## Risks

**Low risk** - Bug fix that only affects async pymongo operations, uses standard Python pattern (`@asynccontextmanager`), and maintains backward compatibility.

### CODE REVIEW

File: ddtrace/contrib/internal/pymongo/async_client.py [hunk 1]
```diff
@@ -95,19 +95,26 @@ async def trace_async_socket_command(func: FunctionType, args: Tuple[Any, ...],
...
 
     socket_instance, dbname, cmd, pin = parsed
-    with trace_cmd(cmd, socket_instance, socket_instance.address) as s:
+    async with async_trace_cmd(cmd, socket_instance, socket_instance.address) as s:
         s, args, kwargs = dbm_dispatch(s, args, kwargs)
         return await func(*args, **kwargs)
 
 
+@contextlib.asynccontextmanager
+async def async_trace_cmd(cmd, socket_instance, address):
```
Comment (brettlangdon) [inline ddtrace/contrib/internal/pymongo/async_client.py:104]: do we have a regression test we can add for this?

Comment (brettlangdon) [reply ddtrace/contrib/internal/pymongo/async_client.py:104]: nvm, I see this now:

> Manually verified. This edge case is difficult to test reliably due to timing requirements and Python version-specific behavior (reported in Python 3.14).

**Review by wantsui — APPROVED** [review]
